libcrux-ml-dsa/src/sample.rs

```diff
@@ -34,20 +34,66 @@ fn rejection_sample_less_than_field_modulus<SIMDUnit: Operations>(
     done
 }
 
-pub(crate) fn sample_four_ring_elements<SIMDUnit: Operations>(
+pub(super) struct SampleArgs<
+    'a,
+    SIMDUnit: Operations,
+    const STACK_SIZE: usize,
+    const ROWS_IN_A: usize,
+    const COLUMNS_IN_A: usize,
+> {
+    pub(super) rand_stack: &'a mut (
+        [u8; STACK_SIZE],
+        [u8; STACK_SIZE],
+        [u8; STACK_SIZE],
+        [u8; STACK_SIZE],
+    ),
+    pub(super) tmp_stack: &'a mut [[i32; 263]],
```

Comment:
I'm experimenting with this one to avoid conversions. (Not for this PR, just FYI.)